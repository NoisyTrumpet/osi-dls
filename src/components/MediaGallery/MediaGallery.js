import React from 'react';
import PropTypes from 'prop-types';
import { Carousel } from '../Carousel';
import { Typography } from '../Typography';
import { List } from '../List';
import { IconButton } from '../IconButton';
import './MediaGallery.scss';

const MediaGallery = ({ children, id }) => {
  const [currentItemIndex, setCurrentItemIndex] = React.useState(0);
  const arrowIDs = {
    next: `${id}-next`,
    prev: `${id}-prev`,
  };

  const handleNewItemShown = (index) => {
    setCurrentItemIndex(index);
  };

  return (
    <div className="media-gallery" id={id}>
      <Carousel enableRewindLooping disableDragging onNewItemVisible={handleNewItemShown} arrows={{ next: `#${arrowIDs.next}`, prev: `#${arrowIDs.prev}` }}>
        {children}
      </Carousel>
      <div className="media-gallery__nav">
        <div className="slide-counter">
          <Typography variant="breadcrumb">{`${currentItemIndex + 1} / ${React.Children.count(children)}`}</Typography>
        </div>
        <div className="carousel__controls-container">
          <List addClass="carousel__controls" horizontal>
            <List.Item>
              <IconButton variant="circular" icon="arrows-tailleft-bold" addClass="carousel__controls-prev" id={arrowIDs.prev}>Previous</IconButton>
            </List.Item>
            <List.Item>
              <IconButton variant="circular" icon="arrows-tailright-bold" addClass="carousel__controls-next" id={arrowIDs.next}>Next</IconButton>
            </List.Item>
          </List>
        </div>
      </div>
    </div>
  );
};

MediaGallery.propTypes = {
  /** MediaGallery items */
  children: PropTypes.node,
  /** Required for arrow buttons to work properly */
  id: PropTypes.string.isRequired,
};

export default MediaGallery;
