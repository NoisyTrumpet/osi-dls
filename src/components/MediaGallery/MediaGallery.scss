.media-gallery {
  .glider-slide {
    .lazy-image-container {
      height: 100%;
      width: 100%;
    }
    .media-gallery_image {
      height: 100%;
      width: 100%;
      object-fit: contain;
    }
  }
}

.media-gallery__nav {
  width: 100%;
  height: $space-32;
  padding: 0 $space-24;
  transform: translateY(-50%);
  .slide-counter {
    display: inline-block;
    background: rgba(255, 255, 255, 0.9);
    text-align: center;
    padding: rem(4px) rem(12px);
    border-radius: $space-16;
    height: $space-32;
    vertical-align: top; /* Notice this line */
  }
  .carousel__controls-container {
    display: block;
    float: right;

    .carousel__controls {
      > .list-item {
        margin-left: $space-8;

        &:first-child {
          margin-left: 0;
        }

        > .btn-icon {
          width: $space-32;
          height: $space-32;
          border-radius: $space-16;
          border-width: 1px;
          background: map-get($colors, 'brand-white');
        }
      }
    }
  }
}