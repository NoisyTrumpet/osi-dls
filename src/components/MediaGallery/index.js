export { default as MediaGallery } from './MediaGallery'; // eslint-disable-line import/prefer-default-export
