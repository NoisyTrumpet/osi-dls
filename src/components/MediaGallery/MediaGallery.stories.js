import React from "react";
import { storiesOf } from "@storybook/react";
import { withSmartKnobs } from "storybook-addon-smart-knobs";
import MediaGallery from "./MediaGallery";
import { Image } from "../Image";

const storyTitle = "components | Carousel/Media Gallery"; // Stories can be organized in a nested structure using ”/” as a separator, and can be given a top-level heading using a ”|” root separator

const parameters = {
  component: MediaGallery,
  componentSubtitle: "<MediaGallery /> defines a Media Gallery",
  jest: ["MediaGallery"]
};

// Pictures from https://osilife.com/cards/greeting-cards/assorted-cards-for-all-occasions-in-floral-organizer-box-box-of-24-1BCE4202.html
storiesOf(storyTitle, module)
  .addParameters(parameters)
  .addDecorator(withSmartKnobs({ ignoreProps: ["children", "id"] }))
  .add("Responsive width w/ Arrow controls", () => (
    <div>
      <MediaGallery id="mg-example-1">
        <Image
          className="media-gallery_image"
          alt="pic-1"
          src="https://cdn-us-ec.yottaa.net/56213828312e5803810043dd/www.hallmark.com/v~4b.21a/dw/image/v2/AALB_PRD/on/demandware.static/-/Sites-hallmark-master/default/dw2586f03e/images/finished-goods/Assorted-Cards-for-AllOccasions-in-Floral-Organizer-Box-Box-of-24-root-1BCE4202_BCE4202_01.jpg_Source_Image.jpg?sw=1024&yocs=27_"
        />
        <Image
          className="media-gallery_image"
          alt="pic-2"
          src="https://cdn-us-ec.yottaa.net/56213828312e5803810043dd/www.hallmark.com/v~4b.21a/dw/image/v2/AALB_PRD/on/demandware.static/-/Sites-hallmark-master/default/dwa35d8bf7/images/finished-goods/Assorted-Cards-for-AllOccasions-in-Floral-Organizer-Box-Box-of-24-root-1BCE4202_BCE4202_02.jpg_Source_Image.jpg?sw=1024&yocs=27_"
        />
        <Image
          className="media-gallery_image"
          alt="pic-3"
          src="https://cdn-us-ec.yottaa.net/56213828312e5803810043dd/www.hallmark.com/v~4b.21a/dw/image/v2/AALB_PRD/on/demandware.static/-/Sites-hallmark-master/default/dw1f0521c8/images/finished-goods/Assorted-Cards-for-AllOccasions-in-Floral-Organizer-Box-Box-of-24-root-1BCE4202_BCE4202_05.jpg_Source_Image.jpg?sw=1024&yocs=27_"
        />
        <Image
          className="media-gallery_image"
          alt="pic-4"
          src="https://cdn-us-ec.yottaa.net/56213828312e5803810043dd/www.hallmark.com/v~4b.21a/dw/image/v2/AALB_PRD/on/demandware.static/-/Sites-hallmark-master/default/dw1e5679a5/images/finished-goods/Assorted-Cards-for-AllOccasions-in-Floral-Organizer-Box-Box-of-24-root-1BCE4202_BCE4202_09.jpg_Source_Image.jpg?sw=1024&yocs=27_"
        />
        <Image
          className="media-gallery_image"
          alt="pic-5"
          src="https://cdn-us-ec.yottaa.net/56213828312e5803810043dd/www.hallmark.com/v~4b.21a/dw/image/v2/AALB_PRD/on/demandware.static/-/Sites-hallmark-master/default/dw7d6a8778/images/finished-goods/Assorted-Cards-for-AllOccasions-in-Floral-Organizer-Box-Box-of-24-root-1BCE4202_BCE4202_13.jpg_Source_Image.jpg?sw=1024&yocs=27_"
        />
        <Image
          className="media-gallery_image"
          alt="pic-6"
          src="https://cdn-us-ec.yottaa.net/56213828312e5803810043dd/www.hallmark.com/v~4b.21a/dw/image/v2/AALB_PRD/on/demandware.static/-/Sites-hallmark-master/default/dwfa5ef493/images/finished-goods/Assorted-Cards-for-AllOccasions-in-Floral-Organizer-Box-Box-of-24-root-1BCE4202_BCE4202_14.jpg_Source_Image.jpg?sw=1024&yocs=27_"
        />
      </MediaGallery>
    </div>
  ))
  .add("Images with different aspect ratios", () => (
    <div>
      <MediaGallery id="mg-example-1">
        <Image
          className="media-gallery_image"
          alt="pic-1"
          src="https://media4.s-nbcnews.com/j/streams/2014/May/140505/2D274905781707-hallmark-two-moms-day-md2926.fit-760w.jpg"
        />
        <Image
          className="media-gallery_image"
          alt="pic-2"
          src="https://www.southernsavers.com/wp-content/uploads/2016/04/free-hallmark-cards.jpg"
        />
        <Image
          className="media-gallery_image"
          alt="pic-6"
          src="https://cdn-us-ec.yottaa.net/56213828312e5803810043dd/www.hallmark.com/v~4b.21a/dw/image/v2/AALB_PRD/on/demandware.static/-/Sites-hallmark-master/default/dwfa5ef493/images/finished-goods/Assorted-Cards-for-AllOccasions-in-Floral-Organizer-Box-Box-of-24-root-1BCE4202_BCE4202_14.jpg_Source_Image.jpg?sw=1024&yocs=27_"
        />
      </MediaGallery>
    </div>
  ))
  .add("Multiple Carousels", () => (
    <div style={{ maxWidth: "400px" }}>
      <MediaGallery id="mg-example-2">
        <Image
          className="media-gallery_image"
          alt="pic-1"
          src="https://cdn-us-ec.yottaa.net/56213828312e5803810043dd/www.hallmark.com/v~4b.21a/dw/image/v2/AALB_PRD/on/demandware.static/-/Sites-hallmark-master/default/dw2586f03e/images/finished-goods/Assorted-Cards-for-AllOccasions-in-Floral-Organizer-Box-Box-of-24-root-1BCE4202_BCE4202_01.jpg_Source_Image.jpg?sw=1024&yocs=27_"
        />
        <Image
          className="media-gallery_image"
          alt="pic-2"
          src="https://cdn-us-ec.yottaa.net/56213828312e5803810043dd/www.hallmark.com/v~4b.21a/dw/image/v2/AALB_PRD/on/demandware.static/-/Sites-hallmark-master/default/dwa35d8bf7/images/finished-goods/Assorted-Cards-for-AllOccasions-in-Floral-Organizer-Box-Box-of-24-root-1BCE4202_BCE4202_02.jpg_Source_Image.jpg?sw=1024&yocs=27_"
        />
        <Image
          className="media-gallery_image"
          alt="pic-3"
          src="https://cdn-us-ec.yottaa.net/56213828312e5803810043dd/www.hallmark.com/v~4b.21a/dw/image/v2/AALB_PRD/on/demandware.static/-/Sites-hallmark-master/default/dw1f0521c8/images/finished-goods/Assorted-Cards-for-AllOccasions-in-Floral-Organizer-Box-Box-of-24-root-1BCE4202_BCE4202_05.jpg_Source_Image.jpg?sw=1024&yocs=27_"
        />
        <Image
          className="media-gallery_image"
          alt="pic-4"
          src="https://cdn-us-ec.yottaa.net/56213828312e5803810043dd/www.hallmark.com/v~4b.21a/dw/image/v2/AALB_PRD/on/demandware.static/-/Sites-hallmark-master/default/dw1e5679a5/images/finished-goods/Assorted-Cards-for-AllOccasions-in-Floral-Organizer-Box-Box-of-24-root-1BCE4202_BCE4202_09.jpg_Source_Image.jpg?sw=1024&yocs=27_"
        />
        <Image
          className="media-gallery_image"
          alt="pic-5"
          src="https://cdn-us-ec.yottaa.net/56213828312e5803810043dd/www.hallmark.com/v~4b.21a/dw/image/v2/AALB_PRD/on/demandware.static/-/Sites-hallmark-master/default/dw7d6a8778/images/finished-goods/Assorted-Cards-for-AllOccasions-in-Floral-Organizer-Box-Box-of-24-root-1BCE4202_BCE4202_13.jpg_Source_Image.jpg?sw=1024&yocs=27_"
        />
        <Image
          className="media-gallery_image"
          alt="pic-6"
          src="https://cdn-us-ec.yottaa.net/56213828312e5803810043dd/www.hallmark.com/v~4b.21a/dw/image/v2/AALB_PRD/on/demandware.static/-/Sites-hallmark-master/default/dwfa5ef493/images/finished-goods/Assorted-Cards-for-AllOccasions-in-Floral-Organizer-Box-Box-of-24-root-1BCE4202_BCE4202_14.jpg_Source_Image.jpg?sw=1024&yocs=27_"
        />
      </MediaGallery>
      <MediaGallery id="mg-example-3">
        <Image
          className="media-gallery_image"
          alt="pic-1"
          src="https://cdn-us-ec.yottaa.net/56213828312e5803810043dd/www.hallmark.com/v~4b.21a/dw/image/v2/AALB_PRD/on/demandware.static/-/Sites-hallmark-master/default/dw2586f03e/images/finished-goods/Assorted-Cards-for-AllOccasions-in-Floral-Organizer-Box-Box-of-24-root-1BCE4202_BCE4202_01.jpg_Source_Image.jpg?sw=1024&yocs=27_"
        />
        <Image
          className="media-gallery_image"
          alt="pic-2"
          src="https://cdn-us-ec.yottaa.net/56213828312e5803810043dd/www.hallmark.com/v~4b.21a/dw/image/v2/AALB_PRD/on/demandware.static/-/Sites-hallmark-master/default/dwa35d8bf7/images/finished-goods/Assorted-Cards-for-AllOccasions-in-Floral-Organizer-Box-Box-of-24-root-1BCE4202_BCE4202_02.jpg_Source_Image.jpg?sw=1024&yocs=27_"
        />
        <Image
          className="media-gallery_image"
          alt="pic-3"
          src="https://cdn-us-ec.yottaa.net/56213828312e5803810043dd/www.hallmark.com/v~4b.21a/dw/image/v2/AALB_PRD/on/demandware.static/-/Sites-hallmark-master/default/dw1f0521c8/images/finished-goods/Assorted-Cards-for-AllOccasions-in-Floral-Organizer-Box-Box-of-24-root-1BCE4202_BCE4202_05.jpg_Source_Image.jpg?sw=1024&yocs=27_"
        />
        <Image
          className="media-gallery_image"
          alt="pic-4"
          src="https://cdn-us-ec.yottaa.net/56213828312e5803810043dd/www.hallmark.com/v~4b.21a/dw/image/v2/AALB_PRD/on/demandware.static/-/Sites-hallmark-master/default/dw1e5679a5/images/finished-goods/Assorted-Cards-for-AllOccasions-in-Floral-Organizer-Box-Box-of-24-root-1BCE4202_BCE4202_09.jpg_Source_Image.jpg?sw=1024&yocs=27_"
        />
        <Image
          className="media-gallery_image"
          alt="pic-5"
          src="https://cdn-us-ec.yottaa.net/56213828312e5803810043dd/www.hallmark.com/v~4b.21a/dw/image/v2/AALB_PRD/on/demandware.static/-/Sites-hallmark-master/default/dw7d6a8778/images/finished-goods/Assorted-Cards-for-AllOccasions-in-Floral-Organizer-Box-Box-of-24-root-1BCE4202_BCE4202_13.jpg_Source_Image.jpg?sw=1024&yocs=27_"
        />
        <Image
          className="media-gallery_image"
          alt="pic-6"
          src="https://cdn-us-ec.yottaa.net/56213828312e5803810043dd/www.hallmark.com/v~4b.21a/dw/image/v2/AALB_PRD/on/demandware.static/-/Sites-hallmark-master/default/dwfa5ef493/images/finished-goods/Assorted-Cards-for-AllOccasions-in-Floral-Organizer-Box-Box-of-24-root-1BCE4202_BCE4202_14.jpg_Source_Image.jpg?sw=1024&yocs=27_"
        />
      </MediaGallery>
    </div>
  ));
