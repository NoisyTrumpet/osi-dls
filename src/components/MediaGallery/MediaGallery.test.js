import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import MediaGallery from './MediaGallery';

describe('<MediaGallery />', () => {
  test('should render', () => {
    const wrapper = shallow(<MediaGallery />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
