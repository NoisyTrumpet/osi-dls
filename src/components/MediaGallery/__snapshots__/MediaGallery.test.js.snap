// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MediaGallery /> should render 1`] = `
<div
  className="media-gallery"
>
  <Carousel
    arrows={
      Object {
        "next": "#undefined-next",
        "prev": "#undefined-prev",
      }
    }
    disableDragging={true}
    duration={0.5}
    enableRewindLooping={true}
    onNewItemVisible={[Function]}
    scrollLockDelay={100}
    slidesToShow={1}
  />
  <div
    className="media-gallery__nav"
  >
    <div
      className="slide-counter"
    >
      <Typography
        variant="breadcrumb"
      >
        1 / 0
      </Typography>
    </div>
    <div
      className="carousel__controls-container"
    >
      <List
        addClass="carousel__controls"
        horizontal={true}
      >
        <ListItem>
          <ForwardRef
            addClass="carousel__controls-prev"
            disabled={false}
            icon="arrows-tailleft-bold"
            id="undefined-prev"
            size={16}
            variant="circular"
          >
            Previous
          </ForwardRef>
        </ListItem>
        <ListItem>
          <ForwardRef
            addClass="carousel__controls-next"
            disabled={false}
            icon="arrows-tailright-bold"
            id="undefined-next"
            size={16}
            variant="circular"
          >
            Next
          </ForwardRef>
        </ListItem>
      </List>
    </div>
  </div>
</div>
`;
